.github/workflows/tests.yaml

```diff
@@ -0,0 +1,273 @@
+            mv /tmp/merged.yml /tmp/config.yml
+          fi
+      - name: Run inclusive naming check
+        uses: canonical/inclusive-naming@main
+        with:
+          fail-on-error: "true"
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          reporter: github-pr-review
+          woke-args: '. -c /tmp/config.yml'
+          filter-mode: nofilter
+          workdir: "./"
+          woke-version: latest
+  shellcheck-lint:
+    name: Shell scripts lint
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - name: Gather files to scan
+        shell: bash
+        id: gather
+        run: |
+          declare -a filepaths
+          shebangregex="^#! */[^ ]*/(env *)?[abk]*sh"
+
+          set -f # temporarily disable globbing so that globs in inputs aren't expanded
+
+          while IFS= read -r -d '' file; do
+          filepaths+=("$file")
+          done < <(find . \
+                        -type f \
+                        '(' \
+                        -name '*.bash' \
+                        -o -name '.bashrc' \
+                        -o -name 'bashrc' \
+                        -o -name '.bash_aliases' \
+                        -o -name '.bash_completion' \
+                        -o -name '.bash_login' \
+                        -o -name '.bash_logout' \
+                        -o -name '.bash_profile' \
+                        -o -name 'bash_profile' \
+                        -o -name '*.ksh' \
+                        -o -name 'suid_profile' \
+                        -o -name '*.zsh' \
+                        -o -name '.zlogin' \
+                        -o -name 'zlogin' \
+                        -o -name '.zlogout' \
+                        -o -name 'zlogout' \
+                        -o -name '.zprofile' \
+                        -o -name 'zprofile' \
+                        -o -name '.zsenv' \
+                        -o -name 'zsenv' \
+                        -o -name '.zshrc' \
+                        -o -name 'zshrc' \
+                        -o -name '*.sh' \
+                        -o -path '*/.profile' \
+                        -o -path '*/profile' \
+                        -o -name '*.shlib' \
+                        ')' \
+                        -print0)
+
+          while IFS= read -r -d '' file; do
+          head -n1 "$file" | grep -Eqs "$shebangregex" || continue
+          filepaths+=("$file")
+          done < <(find . \
+                        -type f ! -name '*.*' -perm /111 \
+                        -print0)
+          echo "filepaths=${filepaths[@]}" >> $GITHUB_OUTPUT
+          set +f # re-enable globbing
+      - if: ${{ steps.gather.outputs.filepaths != '' }}
+        name: Shellcheck Problem Matchers
+        uses: lumaxis/shellcheck-problem-matchers@v1.1.2
+      - if: ${{ steps.gather.outputs.filepaths != '' }}
+        run: shellcheck -f gcc ${{steps.gather.outputs.filepaths}}
+  metadata-lint:
```

Comment:
I guess some of this workflow could be removed since it is charm specific? E.g., we wouldn't expect this repo to have a `metadata.yaml`

Reply:
Indeed that's a leftover, I removed the unnecessary steps